src/transformers/models/altclip/processing_altclip.py

```diff
@@ -35,22 +58,16 @@ class AltCLIPProcessor(ProcessorMixin):
             The image processor is a required input.
         tokenizer ([`XLMRobertaTokenizerFast`], *optional*):
             The tokenizer is a required input.
+        feature_extractor ([`CLIPFeatureExtractor`], *optional*):
+            The feature extractor is a deprecated input.
     """
 
     attributes = ["image_processor", "tokenizer"]
     image_processor_class = "CLIPImageProcessor"
     tokenizer_class = ("XLMRobertaTokenizer", "XLMRobertaTokenizerFast")
 
-    def __init__(self, image_processor=None, tokenizer=None, **kwargs):
-        feature_extractor = None
-        if "feature_extractor" in kwargs:
-            warnings.warn(
-                "The `feature_extractor` argument is deprecated and will be removed in v5, use `image_processor`"
-                " instead.",
-                FutureWarning,
-            )
-            feature_extractor = kwargs.pop("feature_extractor")
-
+    @deprecate_kwarg(old_name="feature_extractor", version="5.0.0", new_name="image_processor")
+    def __init__(self, image_processor=None, tokenizer=None, feature_extractor=None):
```

Comment:
As soon as we added `@deprecate_kwarg(old_name="feature_extractor", version="5.0.0", new_name="image_processor")` we no longer need `feature_extractor` in the function signature, an input keyword argument `feature_extractor` will be renamed to `image_processor`.

```suggestion
    def __init__(self, image_processor=None, tokenizer=None):
```

Reply:
yes indeed, thanks. should be good to go!